### PR TITLE
BLET-44

### PR DESCRIPTION
**Pull request by david-lshift**
This just deals with lots of SonarCloud and SonarLint issues. There's one notable change to ConvertReflectionException - I've removed the fields it captures some context, because the context isn't used anywhere, and they have a nullability issue, which I've made worse by sometimes not specifying the class being converted.

### CODE REVIEW

**Comment by david-lshift** [comment]
I've now ended up doing something quite elaborate to get code coverage with System.exit(int), using the security manager...

**Comment by vlajos** [comment]
It looks good to me, but my java knowledge is a little more limited than to feel comfortable to merge/approve it.

File: src/test/java/net/lshift/spki/suiteb/cli/CliTest.java [hunk 1]
```diff
@@ -64,6 +110,30 @@ public void prettyPrintTest() throws FileNotFoundException, IOException, Invalid
         assertTrue(IOUtils.contentEquals(sPrivate.read(), canonical.read()));
     }
 
+    @Test
+    public void mainWithStdoutTest() throws IOException, ParseException {
```
Comment (tim-shifty) [inline src/test/java/net/lshift/spki/suiteb/cli/CliTest.java]: Not sure what this test is trying to prove. Does it need a comment?

File: src/test/java/net/lshift/spki/suiteb/cli/CliTest.java [hunk 2]
```diff
@@ -64,6 +110,30 @@ public void prettyPrintTest() throws FileNotFoundException, IOException, Invalid
...
+        PrintStream originalOut = System.out;
+        Path workDir = Files.createTempDirectory(this.getClass().getName());
+        workDir.toFile().deleteOnExit();
+        Path signingKeyPath = workDir.resolve("signing-private-key.spki");
+
+        Cli.main(new String [] { "genSigningKey", signingKeyPath.toString() });
```
Comment (tim-shifty) [inline src/test/java/net/lshift/spki/suiteb/cli/CliTest.java]: Why is this here? Does it affect the test result somehow?

Comment (david-lshift) [reply src/test/java/net/lshift/spki/suiteb/cli/CliTest.java]: Added comments to clarify

File: src/test/java/net/lshift/spki/suiteb/cli/CliTest.java (hunk 1 above)
Comment (david-lshift) [reply src/test/java/net/lshift/spki/suiteb/cli/CliTest.java]: added comments to clarify
